packages/peregrine/lib/talons/CartPage/PriceAdjustments/useCouponCode.js

```diff
@@ -10,12 +10,14 @@ export const useCouponCode = props => {
     } = props;
 
     const [{ cartId }] = useCartContext();
-    const [fetchAppliedCoupons, { data, error: fetchError }] = useLazyQuery(
-        getAppliedCouponsQuery,
-        {
-            fetchPolicy: 'cache-and-network'
-        }
-    );
+    // Create a memoized error map and toggle individual errors when they change
+    const errors = useMemo(() => new Map(), []);
```

Comment:
I think the suggestion from @tjwiebell is necessary, actually.

The approach used here is a mutable one: it memoizes the map and then mutates it, so components are vulnerable to missing the update entirely. Instead, we want an immutable approach, where a new map is created and returned whenever the underlying data changes.

Reply:
Handled in c787ebf